src/ngraph/ngraph_utils.h

```diff
@@ -271,6 +271,50 @@ inline std::size_t hash_combine(const std::size_t& seed, const T& val) {
 void dump_graph(std::shared_ptr<ngraph::Function> f, std::string src_loc = "",
                 std::string filename_suffix = "");
 
+// We define the term "vector plus axes" to refer to tensor shapes meeting the following criteria:
```

Comment:
Nitpick: "vector plus axes" takes a while to digest. Would vector_with_expanded_dims make more sense? 

I get what you're doing here, and it's good to abstract it, I just find the naming a little confusing. Naming, it's hard.

Reply:
Sounds good. Honestly not sure what's best, just reporting my initial confusion.